Netbeans, integration tests, Docker

Now that we've merged...

- #9273

.... Netbeans users see `Caused by: java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration` if they try to build and don't have Docker running.

From the [thread](https://iqss.slack.com/archives/C010LA04BCG/p1696448584278029) in Slack, @poikilotherm suggests reconfiguring Netbeans like this:

![image](https://github.com/IQSS/dataverse/assets/21006/23739f90-9412-4c53-9d18-15249fbefa07)

We should try it out and make a PR to update a future versions of https://guides.dataverse.org/en/6.0/developers/classic-dev-env.html#build-the-dataverse-software-war-file above where we say to click build in Netbeans:

![Screenshot 2023-10-04 at 5 14 13 PM](https://github.com/IQSS/dataverse/assets/21006/45339c64-3ac1-41b7-b619-9e1e9cebd897)

--- a/doc/sphinx-guides/source/developers/testing.rst
+++ b/doc/sphinx-guides/source/developers/testing.rst
@@ -316,7 +316,7 @@ Please make sure to:
    .. code:: java
 
        /** A very minimal example for a Testcontainers integration test class. */
-       @Testcontainers
+       @Testcontainers(disabledWithoutDocker = true)
        @Tag(edu.harvard.iq.dataverse.util.testing.Tags.INTEGRATION_TEST)
        @Tag(edu.harvard.iq.dataverse.util.testing.Tags.USES_TESTCONTAINERS)
        class MyExampleIT { /* ... */ }

--- a/src/test/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OIDCAuthenticationProviderFactoryIT.java
+++ b/src/test/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OIDCAuthenticationProviderFactoryIT.java
@@ -54,7 +54,7 @@ import static org.mockito.Mockito.when;
 
 @Tag(Tags.INTEGRATION_TEST)
 @Tag(Tags.USES_TESTCONTAINERS)
-@Testcontainers
+@Testcontainers(disabledWithoutDocker = true)
 @ExtendWith(MockitoExtension.class)
 // NOTE: order is important here - Testcontainers must be first, otherwise it's not ready when we call getAuthUrl()
 @LocalJvmSettings

